src/main/java/organice/logic/commands/FindCommand.java

```diff
@@ -1,44 +1,154 @@
+import static organice.logic.parser.CliSyntax.PREFIX_PHONE;
+import static organice.logic.parser.CliSyntax.PREFIX_TYPE;
 
-import organice.commons.core.Messages;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import javafx.collections.FXCollections;
+import javafx.collections.transformation.FilteredList;
+
+import organice.logic.parser.ArgumentMultimap;
 import organice.model.Model;
+import organice.model.person.Name;
+import organice.model.person.Person;
 import organice.model.person.PersonContainsPrefixesPredicate;
 
+
 /**
  * Finds and lists all persons in address book whose prefixes match any of the argument prefix-keyword pairs.
+ * Performs fuzzy searching based on Levenshtein Distance.
  * Keyword matching is case insensitive.
  */
 public class FindCommand extends Command {
 
     public static final String COMMAND_WORD = "find";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose prefixes match any of "
-            + "the specified prefix-keywords pairs (case-insensitive) and displays them as a list with index numbers.\n"
-            + "List of Prefixes: n/, ic/, p/, a/, t/, pr/, b/, d/, tt/, exp/, o/"
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose prefixes are similar to any of"
+            + " the specified prefix-keywords pairs (case-insensitive) and displays them as a list with index numbers."
+            + "\nList of Prefixes: n/, ic/, p/, a/, t/, pr/, b/, d/, tt/, exp/, o/"
             + "Parameters: PREFIX/KEYWORD [MORE_PREFIX-KEYWORD_PAIRS]...\n"
             + "Example: " + COMMAND_WORD + " n/alice t/doctor";
 
-    private final PersonContainsPrefixesPredicate predicate;
+    // Maximum Levenshtein Distance tolerated for a fuzzy match
+    private static final int FUZZY_THRESHOLD = 5;
+
+    private final ArgumentMultimap argMultimap;
+
+    public FindCommand(ArgumentMultimap argMultimap) {
+        this.argMultimap = argMultimap;
+    }
+
+    /**
+     * Returns a copy of the {@code inputList} which is filtered according to maximum tolerable Levenshtein Distance
+     * (edit distance) then sorted according to ascending level of distance.
+     */
+    private List<Person> fuzzyMatch(ArgumentMultimap argMultimap, List<Person> inputList) {
+        // Fuzzy Match by Levenshtein Distance is not implemented for following prefixes:
+        // Age, Priority, BloodType, TissueType
+        // Typos in these fields have a very small Levenshtein Distance (LD) as typical field length is very small
+
+        List<String> nameKeywords = argMultimap.getAllValues(PREFIX_NAME);
+        List<String> nricKeywords = argMultimap.getAllValues(PREFIX_NRIC);
+        List<String> phoneKeywords = argMultimap.getAllValues(PREFIX_PHONE);
+        List<String> typeKeywords = argMultimap.getAllValues(PREFIX_TYPE);
```

Comment:
As discussed earlier, you should type validate.

Reply:
Another PR